apps/onboarding/src/session/session.service.ts

```diff
@@ -51,4 +66,15 @@ export class SessionService {
       return existingSession
     }
   }
+
+  async checkSessionQuota(session: Session, quota: SessionQuota, updateUsage: boolean = true) {
+    const usage = session.checkQuota(quota, updateUsage)
+    if (updateUsage) {
+      await this.update(session.id, session)
```

Comment:
Like I said before I would remove the update from the `session.checkQuota`. And handle the update here.